plugins/modules/vmware_vmkernel.py

```diff
@@ -792,7 +787,7 @@ def find_dvspg_by_key(self, dv_switch, portgroup_key):
 
         return None
 
-    def set_vsan_service_type(self):
+    def set_vsan_service_type(self, enable_vsan=False):
```

Comment:
```suggestion
    def set_vsan_service_type(self, enable_vsan):
```

I don't think you'll need a default here since `set_vsan_service_type` isn't called without setting  enable_vsan`.

Reply:
Right, I just thought it would be proper manners to keep the calling convention of the function "just in case". Just applied a fix.